fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+import com.starrocks.analysis.TableName;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.Pair;
+import com.starrocks.common.util.concurrent.lock.LockType;
+import com.starrocks.common.util.concurrent.lock.Locker;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.MetadataMgr;
+import com.starrocks.sql.ast.AstTraverser;
+import com.starrocks.sql.ast.DeleteStmt;
+import com.starrocks.sql.ast.InsertStmt;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.StatementBase;
+import com.starrocks.sql.ast.TableRelation;
+import com.starrocks.sql.ast.UpdateStmt;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * QueryLocker is used to obtain the lock corresponding to the metadata used in Query
+ * Will acquire the read locks of all tables involved in the query
+ * and the intention read locks of the database where the tables are located.
+ * <p>
+ * In terms of compatibility, when Config.use_lock_manager is turned off,
+ * it will be consistent with the original db-lock logic
+ * and obtain the db-read-lock of all dbs involved in the query.
+ */
+public class QueryLocker {
```

Comment:
```suggestion
public class PlannerMetaLocker {
```

reason: not only used by query, and always used as a param of StatementPlanner.lock() to acquire lock on the meta being used. BTW, comments should also be updated.